### ISSUE TITLE
Talk back is not reading Arabic language in Audio Language dialog box 

### ISSUE THREAD

**Issue by KolliAnitha** [oppia/oppia-android#4312]
**Describe the bug**
Talk back is not reading Arabic language in Audio Language dialog box 

**To Reproduce**
1. Go to 'Place Values'
2. Double tap on 'What are place values'
3. Double tap on 'Audio' button
4. Double tap on 'Change language' icon
5. Audio Language dialog box appears
6. Swipe across the various languages

**Expected behavior**
Talk back to read all the language names including Arabic

**Environment**
Device name: One plus Nord2 5G
Android version : Android 11
App version : 0.7-alpha-d1b2827517

**Explanation and suggested solution**
In `LanguageDialogFragment`, we use ` appLanguageResourceHandler.computeLocalizedDisplayName(audioLanguage)` to get the display name for ecah language. The Arabic language therefore is displayed in Arabic characters, which talkback, when localized to English, cannot read. 

Talkback relies on content descriptions where available, but since this list is dynamic, we can use an accessibility delegate to instruct talkback on how to read Arabic(and Hindi too).

When creating the dialog, override the accessibility properties of the list items by accessing the AlertDialog's ListView after creation:

```kotlin
val options = languageNameArrayList.toTypedArray<CharSequence>()
val languageInterface: LanguageInterface = parentFragment as AudioFragment

val alertDialog = AlertDialog.Builder(ContextThemeWrapper(activity as Context, R.style.OppiaDialogFragmentTheme))
  ...
    .create()

alertDialog.setOnShowListener {
    val listView = alertDialog.listView
    for (i in 0 until listView.childCount) {
        val item = listView.getChildAt(i)
        item?.setAccessibilityDelegate(object : View.AccessibilityDelegate() {
            override fun onInitializeAccessibilityNodeInfo(
                host: View?,
                info: AccessibilityNodeInfo?
            ) {
                super.onInitializeAccessibilityNodeInfo(host, info)
                // Get the item text
           
                // Set content description for TalkBack
                        }
        })
    }
}

return alertDialog

```

UI tests are expected for the submitted code.

**Comment by BenHenning** [comment]
For searching context, this issue was found in 0.7-alpha (MR4).

**Comment by adhiamboperes** [comment]
@Vishwajith-Shettigar, could you please test and leave a comment whether this issue is still occurring in the playstore version of the app?
 
 You could also test with the development version on the develop branch, using the fractions topic. @MohitGupta121  should be able to help in case you have any questions

**Comment by MohitGupta121** [comment]
@adhiamboperes yes I think this issue is already fixed, I checked and found Talkback reading Arabic language text. 

https://github.com/oppia/oppia-android/assets/76530270/e282d114-4037-4b7b-9f14-931261a90986

**Comment by adhiamboperes** [comment]
Hi @MohitGupta121, once you get the production assets set up, please verify this with the correct lesson per the issue description, thanks!

**Comment by MohitGupta121** [comment]
@adhiamboperes yes you are right there is a issue, here is updated screen recording:


https://github.com/oppia/oppia-android/assets/76530270/de39e595-e467-4a6c-944b-fcd3f47031c6

**Comment by adhiamboperes** [comment]
@elin4046, you can work on this

**Comment by elin4046** [comment]
Hi @adhiamboperes, this is the approach that I'm planning to take:

**Issue** 
Currently, Talkback will read the list of languages using the device's configured TTS language setting (see screenshot for this setting). For devices where this setting is not set to Arabic, this results in any Arabic words being skipped.

![Screenshot 2024-07-12 at 12 38 19 PM](https://github.com/user-attachments/assets/17b33f5d-1e10-4ab2-b33b-042bbd90f4c9)

Additionally, the current behavior means that the other languages in the list _aren't_ being read in their native TTS languages, which can be confusing to understand (eg. the Chinese characters for Mandarin Chinese are read in a hard-to-understand English TTS 'accent'). 

**Proposed Approach**
Update `AppLanguageResourceHandler::computeLocalizedDisplayName` so that it returns a `SpannableString` instead of a `String` and wrap the returned string with `LocaleString.` By doing so, we can specify a locale that allows TalkBack to dyanmically switch to the proper TTS voice based on the language. This change would also require updating the `OptionsAudioLanguageViewModel` and `AudioLanguageSelectionViewModel` display language property to use `SpannableString` instead of `String.` 

Let me know if these are the methods/files that should be updated in this fix, thanks!

**Comment by adhiamboperes** [comment]
Hi @elin4046, thanks for researching this. I have a few follow ups:
- Typically, we don't use Locale directly in the codebase. However, if you can create a function in  the `OppiaLocale` class, it might work.
- Configuring locale specific tts voice is technically out of scope for this issue. However, if it is an automatic side effect of the fix, then we can test it out and get product feedback on it. 


Do you mind  trying this fix and doing a screen recording so we can see what it is like?

**Additional Context**
The Oppia app is set up to use the device language by default, and if the device is configure in mandarin for example, most tts will be in mandarin. However, for app strings, mandarin is not one of the translated languages we currently support, and talkback relies on content descriptions to describe the app, so for untranslated strings will default to English.

**Comment by adhiamboperes** [comment]
@theayushyadav11, here's another issue to look at.

**Comment by theayushyadav11** [comment]
Hey @adhiamboperes,thanks for considering me for this work.
I’m currently unable to reproduce the **_Place Values_** screen on the app installed via Android Studio, even though it’s accessible in the version downloaded from the Play Store. Could you kindly guide me on how to test this screen using the app installed through Android Studio?

**Comment by adhiamboperes** [comment]
@theayushyadav11, thanks for letting me know. The lesson is currently not available on the developer version. We may have to test this using a differnt lesson like Fractions or Ratios.

**Comment by theayushyadav11** [comment]
Hi @adhiamboperes , currently there is no option available for arabic language in Fractions or Ratios.

**Comment by adhiamboperes** [comment]
Thanks @theayushyadav11. It seems that this issue is currently blocked. It may be best to work on a different issue until we can reproduce this on the development environment
Thanks for taking a look!

**Comment by adhiamboperes** [comment]
@subhajitxyz, is this a different issue from the one you fixed in https://github.com/oppia/oppia-android/pull/5613?

**Comment by subhajitxyz** [comment]
> [@subhajitxyz](https://github.com/subhajitxyz), is this a different issue from the one you fixed in [#5613](https://github.com/oppia/oppia-android/pull/5613)?

Yes. This is a different issue.

**Comment by EskAlam06** [comment]
The Arabic Language is not available on Audio Language Dialog Box-

![Image](https://github.com/user-attachments/assets/921ed23e-5b33-4fd7-86e5-5dbe4848b00d)

**Comment by adhiamboperes** [comment]
This issue needs to be reproduced with a lesson that is known to have Arabic voiceovers. The dev assets do not have this.
